File: mergeAlternately.ts
function mergeAlternately(word1: string, word2: string): string {
  const a = word1.split("");
  const b = word2.split("");
  const max = Math.max(a.length, b.length);
  return Array.from({ length: max }).map((_, index) => (a[index] ?? '') + (b[index] ?? '')).join("");
};

const a = (mergeAlternately("abc", "xyz"))
const b = (mergeAlternately("ab", "xyz"))
const c = (mergeAlternately("abc", "xy"))

console.log(a, a === "axbycz")
console.log(b, b === "axbyz")
console.log(c, c === "axbyc")

File: src-py/mergeAlternately.py
str1 = "abc"
str2 = "xyz"


class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        def gen():
            minL: int = min(len(word1), len(word2))
            for i in range(minL):
                yield word1[i]
                yield word2[i]
            yield from word1[minL:]
            yield from word2[minL:]
        return "".join(gen())


print(Solution().mergeAlternately(str1, str2))
